zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -764,11 +764,19 @@ private static void sendToKafka(Collection collection) throws IOException {
             List<String> datasetUris = collection.getDatasetVersionDetails()
                     .stream().map(content -> convertUriForEvent(content.uri))
                     .collect(Collectors.toList());
+
+            // valid uris for published CMD versions of a dataset (edition) - /dataset/{datatsetId}/editions/{edition}/versions/{version}/metadata
+            List<String> validDatasetUris = new ArrayList<>();
```

Comment:
Rather than creating a new array, why not use the already existing stream above to filter out the unwanted URI's before the stream is collected to a list?
You can then have a separate validation function which can be cleanly unit tested. See section 2.1 in this tutorial for an example. https://www.baeldung.com/java-stream-filter-lambda